ts/resolution.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2014 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Matt Tracy (matt.r.tracy@gmail.com)
+
+package ts
+
+import (
+	"fmt"
+	"time"
+)
+
+// Resolution is used to enumerate the different resolution values supported by
+// cockroach.
+type Resolution int64
+
+const (
+	// Resolution10s stores data with a sample resolution of 10 seconds.
+	Resolution10s Resolution = iota
+)
+
+// sampleDurationByResolution is a map used to retrieve the sample duration
+// corresponding to a Resolution value. Sample durations are expressed in
+// nanoseconds.
+var sampleDurationByResolution = map[Resolution]int64{
+	Resolution10s: int64(time.Second * 10),
+}
+
+// keyDurationByResolution is a map used to retrieve the key duration
+// corresponding to a Resolution value; the key duration determines how many
+// samples are stored at a single cockroach key. Sample durations are expressed
+// in nanoseconds.
+var keyDurationByResolution = map[Resolution]int64{
+	Resolution10s: int64(time.Hour),
+}
+
+// SampleDuration returns the sample duration corresponding to this resolution
+// value, expressed in nanoseconds.
+func (r Resolution) SampleDuration() int64 {
```

Comment:
Not sure if `SampleDuration` or `KeyDuration` should be exported, but ok to leave as-is for now.